x-pack/plugins/security_solution/public/resolver/store/selectors.ts

```diff
@@ -122,3 +132,93 @@ function composeSelectors<OuterState, InnerState, ReturnValue>(
 ): (state: OuterState) => ReturnValue {
   return (state) => secondSelector(selector(state));
 }
+
+const spatiallyIndexedEntities = createSelector(
```

Comment:
Since this only needs stuff from the `data` selectors, I think this should be moved to the `data` selectors. And  `processNodePositionsAndEdgeLineSegments` can probably be made private. Nothing outside of `data` needs to be exposed to the underlying structure of the graph. The interface can just be
```
(query: AABB) => Entities[]
```